src/schema-generation/order-by-and-pagination-augmentation.ts

```diff
@@ -267,10 +337,46 @@ export class OrderByAndPaginationAugmentation {
                             ? and(filterNode, paginationFilter)
                             : filterNode || paginationFilter,
                 });
+
+                if (
+                    options?.firstLimitCheckType === LimitTypeCheckType.RESULT_VALIDATOR &&
+                    options?.operation &&
+                    args[FIRST_ARG] === undefined &&
+                    maxCount
+                ) {
+                    return this.wrapWithImplicitListTruncationResultValidator(
+                        transformListNode,
+                        maxCount,
```

Comment:
instead of passing `implicitLimitForRootEntityQueries + 1` here and make the function subtract one again, I think it would be more striaghtforward to just pass `info.implicitLimitForRootEntityQueries` as the limit here

Reply:
if the query/mutation already contains a limit, we don't need the implicit limit at all. Therefore we also don't need a validator. The implicit limit and validator are only needed if no limit is in place. I think this would also be my reply to the [question above](https://github.com/AEB-labs/cruddl/pull/318#discussion_r1674303598).

Reply:
Ah yes you are right 🙈 👍